core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/udf/java/JavaRuntimeCompilation.scala

```diff
@@ -0,0 +1,70 @@
+package edu.uci.ics.texera.workflow.operators.udf.java
+
+import java.io.ByteArrayOutputStream
+import java.net.URI
+import java.util
+import javax.tools._
+
+object JavaRuntimeCompilation {
+  val compiler: JavaCompiler = ToolProvider.getSystemJavaCompiler
+
+  def compileCode(code: String): Class[_] = {
+    val packageName = "edu.uci.ics.texera.workflow.operators.udf.java"
+
+    //to hide it from user we will append the package in the udf code.
+    val codeToCompile = s"package $packageName;\n$code"
+    val defaultClassName = s"$packageName.JavaUDFOpExec"
+
+    // TODO: test if we can use the same fileManager across different compilations
+    val fileManager: CustomJavaFileManager = new CustomJavaFileManager(
+      compiler.getStandardFileManager(null, null, null)
+    )
+
+    compiler
+      .getTask(
+        null,
```

Comment:
compilation does report error! if you provide a writer here on this argument, it will write the error out. By default, the error is only written to sys.err. I think we want to get this error and display to user. please try if you can get the error and display it to user. if not, that's also fine.

Reply:
Done